test/plugins/collections.test.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const assert = require('chai').assert;
+const sinon = require('sinon');
+const hapi = require('hapi');
+const mockery = require('mockery');
+const urlLib = require('url');
+const hoek = require('hoek');
+const testCollection = require('./data/collection.json');
+
+sinon.assert.expose(assert, { prefix: '' });
+
+const getCollectionMock = (collection) => {
+    const mock = hoek.clone(collection);
+
+    mock.update = sinon.stub();
+    mock.toJson = sinon.stub().returns(collection);
+    mock.remove = sinon.stub();
+
+    return mock;
+};
+
+const getUserMock = (user) => {
+    const mock = hoek.clone(user);
+
+    return mock;
+};
+
+// const collectionsGetterMock = collections => Promise.resolve(collections);
+
+describe('collection plugin test', () => {
+    const username = 'jsequeira';
+    const userId = testCollection.userId;
+    let collectionFactoryMock;
+    let userFactoryMock;
+    let collectionMock;
+    let userMock;
+    let plugin;
+    let server;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach((done) => {
+        collectionFactoryMock = {
+            create: sinon.stub(),
+            get: sinon.stub()
+        };
+        userFactoryMock = {
+            get: sinon.stub()
+        };
+
+        collectionMock = getCollectionMock(testCollection);
+        collectionMock.remove.resolves(null);
+        collectionMock.update.resolves(collectionMock);
+        collectionFactoryMock.create.resolves(collectionMock);
+
+        userMock = getUserMock({
+            username,
+            id: userId
+        });
+        userFactoryMock.get.withArgs({ username }).resolves(userMock);
+
+        /* eslint-disable global-require */
+        plugin = require('../../plugins/collections');
+        /* eslint-enable global-require */
+        server = new hapi.Server();
+        server.app = {
+            collectionFactory: collectionFactoryMock,
+            userFactory: userFactoryMock
+        };
+        server.connection({
+            port: 1234
+        });
+
+        server.auth.scheme('custom', () => ({
+            authenticate: (request, reply) => reply.continue({})
+        }));
+        server.auth.strategy('token', 'custom');
+        server.auth.strategy('session', 'custom');
+
+        return server.register([{
+            register: plugin
+        }], done);
+    });
+
+    afterEach(() => {
+        server.stop();
+        server = null;
+        mockery.deregisterAll();
+        mockery.resetCache();
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('registers the plugin', () => {
+        assert.isOk(server.registrations.collections);
+    });
+
+    describe('POST /collections', () => {
```

Comment:
I discussed that with @minz1027 earlier and we came to the conclusion that the since the Model is mocked, the only way to test that functionality out is in the data schema and data store end and not the API directly.